File: android/src/main/kotlin/com/example/xplayer/models/XPlayerValue.kt
// To parse the JSON, install kotlin's serialization plugin and do:
//
// val json         = Json { allowStructuredMapKeys = true }
// val xPlayerValue = json.parse(XPlayerValue.serializer(), jsonString)

package com.example.xplayer.models

import kotlinx.serialization.Serializable

@Serializable
data class XPlayerValue (
     val playerState: Int = 1,
     val position: Long = 0L,
     val bufferedPosition: Long = 0L,
     val playbackSpeed: Float = 0F,
     val qualities: List<Quality> = mutableListOf()
)

File: ios/Classes/Models/XPlayerValue.swift
//
//  XPlayerValue.swift
//  xplayer
//
//  Created by Sovanreach on 14/1/24.
//

import Foundation

class XPlayerValue: NSObject, Codable {
    
    var playerState: Int = 1
    var position: Int = 0
    var bufferedPosition: Int=0
    var playbackSpeed: Float = 0.0
    var qualities: Array<Quality> = []
    
    override init() {
    }
    
    init(playerState: Int = 1, position: Int = 0, bufferedPosition: Int = 0 , playbackSpeed: Float = 0.0, qualities: Array<Quality> = []) {
        self.playerState = playerState
        self.position = position
        self.bufferedPosition = bufferedPosition
        self.playbackSpeed = playbackSpeed
        self.qualities = qualities
    }

}
